### DISCUSSION TITLE
How to tell if an activity is followers only or everyone?

### DISCUSSION THREAD

**Discussion by henriwoodcock** [hozn/stravalib#587] [Q&A] [answered]
Hi I submitted an app to strava and they said I need to respect user's privacy. The app makes a public leaderboard of users submitting to specific segments. I have changed my app to use the scope `activity:read` instead of `activity:read_all`, but I have no way of knowing whether an activity is public or followers only. This is also true in the webhook. What's the best way to do this using the stravalib client?

**Comment by jsamoocha** [comment, marked as answer]
Hi, it should be the `DetailedActivity.visibility` attribute. It is undocumented by Strava but included in responses and part of our model.

**Reply by henriwoodcock** [reply]
thanks, I was running version 1.7, have just upgraded
